### PR TITLE
Safeloader: respect disable on all Python tests and ignore teststmpdir

### PR DESCRIPTION
**Pull request by clebergnu**
`avocado.Test.teststmpdir` property is badly named because it matches what is expected to be a test.
    
Because Avocado inherits from Python's `unittest.TestCase`, and the extended coverage for corner cases in the safeloader introduced, along with the characteristic of the resolver that will give other plugins a chance, this property is now being found as a Python unittest test method.

These changes prevent that from happening, resulting in a more straightforward resolution.  That is, the `python-unittest` resolver plugin will never find tests in a `avocado.Test` based class that contains a `:avocado: disable` tag. 

Besides this change, a method rename is desirable, but this will cause API compatibility issues, which we guarantee until a new LTS release. So, for now, let's disable all test* methods on avocado.Test from being recognized as Python unittest methods.

### CODE REVIEW

File: avocado/core/test.py [hunk 1]
```diff
@@ -215,6 +215,8 @@ class Test(unittest.TestCase, TestData):
 
     You'll inherit from this to write your own tests. Typically you'll want
     to implement setUp(), test*() and tearDown() methods on your own tests.
+
+    :avocado: disable
```
Comment (willianrampazzo) [inline avocado/core/test.py:219]: Oh boy :D

**Review by willianrampazzo — APPROVED** [review]

**Review by beraldoleal — APPROVED** [review]
LGTM
